Plugins/Encounter/Encounter.cpp

```diff
@@ -278,4 +281,95 @@ ArgumentStack Encounter::GetCurrentNumSpawned(ArgumentStack&& args)
     return Services::Events::Arguments(retVal);
 }
 
+ArgumentStack Encounter::GetEncounterGeometry(ArgumentStack&& args)
+{
+    std::string retVal;
+
+    if (auto *pObject = encounter(args))
+    {
+        if (auto *pEncounter = Utils::AsNWSEncounter(pObject))
+        {
+            retVal.reserve(32 * pEncounter->m_nNumActivateVertices);
+
+            for(int i = 0; i < pEncounter->m_nNumActivateVertices; i++)
+            {
+                retVal += "{" + std::to_string(pEncounter->m_pvActivateVertices[i].x) + ", " +
+                                std::to_string(pEncounter->m_pvActivateVertices[i].y) + ", " +
+                                std::to_string(pEncounter->m_pvActivateVertices[i].z) + "}";
+            }
+        }
+        else
+        {
+            LOG_WARNING("NWNX_Encounter_GetEncounterGeometry() called on non encounter object.");
+        }
+    }
+
+    return Services::Events::Arguments(retVal);
+}
+
+ArgumentStack Encounter::SetEncounterGeometry(ArgumentStack&& args)
+{
+    if (auto *pObject = encounter(args))
+    {
+        const auto sGeometry = Services::Events::ExtractArgument<std::string>(args);
+
+        if (auto *pEncounter = Utils::AsNWSEncounter(pObject))
+        {
+            auto str = sGeometry.c_str();
+            std::vector<Vector> vecVerts;
+            Vector vec = {};
+
+            do {
+                int cnt = std::sscanf(str, "{%f,%f,%f", &vec.x, &vec.y, &vec.z);
+
+                if (cnt != 2 && cnt != 3)
+                {
+                    LOG_NOTICE("NWNX_Encounter_SetEncounterGeometry() invalid geometry string at: %s", str);
+                    break;
+                }
+
+                if (cnt == 2)
+                    vec.z = pEncounter->GetArea()->ComputeHeight(vec);
+
+                vecVerts.push_back(vec);
+            } while ((str = std::strstr(str + 1, "{")));
+
+            if (vecVerts.size() > 2)
+            {
+                CNWSArea *pArea = pEncounter->GetArea();
+                pEncounter->RemoveFromArea();
+
+                if (pEncounter->m_pvActivateVertices)
```

Comment:
I get that this is a clone of Object.cpp code, but these `if`s are useless, just delete them. `delete nullptr` is guaranteed to do nothing.